File: frontend/src/types/cliente.ts
export type Cliente = {
  id: string;
  nome: string;
  cpf?: string | null;
  telefone?: string | null;
  email?: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
};

export type ClienteCreate = {
  nome: string;
  cpf?: string;
  telefone?: string;
  email?: string;
};

export type ClienteListResponse = {
  items: Cliente[];
  limit: number;
  offset: number;
  total: number;
};

File: backend/app/models/cliente.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, Uuid, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base


class Cliente(Base):
    __tablename__ = "clientes"

    id: Mapped[uuid.UUID] = mapped_column(
        Uuid(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
    )
    nome: Mapped[str] = mapped_column(String(255), nullable=False)
    cpf: Mapped[str | None] = mapped_column(
        String(14),
        unique=True,
        index=True,
        nullable=True,
    )
    telefone: Mapped[str | None] = mapped_column(String(30), nullable=True)
    email: Mapped[str | None] = mapped_column(String(255), nullable=True)
    ativo: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        server_default=func.now(),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        server_default=func.now(),
        onupdate=func.now(),
    )

    nfc_tags = relationship(
        "NFCTag",
        back_populates="cliente",
        cascade="all, delete-orphan",
    )
    leituras_nfc = relationship("LeituraNFC", back_populates="cliente")
